Drv/LinuxI2CDriver/LinuxI2CDriverComponentImpl.cpp

```diff
@@ -0,0 +1,90 @@
+// ======================================================================
+// \title  LinuxI2CDriverComponentImpl.cpp
+// \author aaron
+// \brief  cpp file for LinuxI2CDriver component implementation class
+//
+// \copyright
+// Copyright 2009-2015, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+
+
+#include <Drv/LinuxI2CDriver/LinuxI2CDriverComponentImpl.hpp>
+#include "Fw/Types/BasicTypes.hpp"
+
+#define I2C_FILE (char*)"/dev/i2c-1"
+
+int init_address(int addr) {
+  int fd = open(I2C_FILE, O_CLOEXEC | O_RDWR);
```

Comment:
This open call appears to never be closed.  Also why `O_CLOEXEC`?

Reply:
Also, are you required to open the file anew for every call, or can it be opened once and shared with all other callers.

Reply:
I chose to use O_CLOEXEC because this means that it will be closed on the execution of a read or write operation, so it is intended to be used once. Is it preferred that an i2c address be opened, then used for transactions, and then closed? If so, I could switch to this flow and remove the O_CLOEXEC, but I thought that this simplified the API in a very nice way.

Reply:
This is an older PR, but we should make the interface consistent with the other drivers. The pattern there is:

```
open(filename)
read...
write...
destructor closes file
```
I don't think the file name should be hard coded in case a linux flavor uses a different file name.

Having the port guarded allows the `ioctl` call and then the `read`/`write` to be atomic.